crates/extensions/c8y_mapper_ext/src/service_monitor.rs

```diff
@@ -42,16 +41,20 @@ pub fn convert_health_status_message(
         health_status = "unknown".into();
     }
 
-    // FIXME: `ancestors_external_ids` gives external ids starting from the parent, but for health
-    // we need XID of current device as well
-    let mut external_ids = vec![entity.external_id.as_ref().to_string()];
-    external_ids.extend_from_slice(ancestors_external_ids);
-    let status_message =
-        smartrest::inventory::service_status_update_message(&external_ids, &health_status);
-
-    mqtt_messages.push(status_message);
+    let display_name = entity
+        .other
+        .get("name")
+        .and_then(|v| v.as_str())
+        .or(entity.topic_id.default_service_name())
+        .unwrap_or(entity.external_id.as_ref());
+    let Ok(status_message) =
+        // smartrest::inventory::service_status_update_message(&external_ids, &health_status);
+        smartrest::inventory::service_creation_message(entity.external_id.as_ref(), display_name, entity.r#type.as_str(), &health_status, ancestors_external_ids) else {
```

Comment:
```suggestion
        smartrest::inventory::service_creation_message(entity.external_id.as_ref(), display_name, display_type, &health_status, ancestors_external_ids) else {
```
The service type also needs to be derived from `entity.other.get("type")` like you've done for `display_name`.

Reply:
You're right, I confused the types there.